yarn.lock

```diff
@@ -1161,6 +1161,13 @@
   dependencies:
     "@sinonjs/commons" "^1.7.0"
 
+"@sinonjs/fake-timers@^8.0.1":
```

Comment:
Q: Is there a particular feature this dependency has that made you choose it over jest's fake timers?

Reply:
Good question. The functionality that is being tested inside of GasFeeController involves asynchronous behavior via not only `setTimeout` but also via promises. That is, inside of the `setTimeout`, we run one or more promises, wait for all to finish, then schedule the next `setTimeout`, then run more promises, then schedule the next `setTimeout`, etc. I tried Jest's fake timers at first and ran into some perplexing timing issues. Although Jest was effective in stubbing out `setTimeout`, it seemed that the promises "leaked out" of the setTimeout and were therefore running after key assertion lines in the tests ran. This made it impossible to truly test the behavior in question. I did some research and it seems that Jest's fake timers have had [issues](https://stackoverflow.com/questions/51126786/jest-fake-timers-with-promises) in the past with this sort of thing. In Jest 26+, you can supposedly say `jest.useFakeTimers('modern')` to switch over to a different implementation [which makes use of Sinon's fake timers](https://github.com/facebook/jest/pull/7776), but I tried this and it seemed to not know what I was talking about. Plus, this "modern" implementation [seems to have some quirks too](https://github.com/facebook/jest/issues/10221#issuecomment-654687396) that seems to indicate that Jest makes some modifications to Sinon's timers. In the end, using Sinon directly solved the problem.

Reply:
This is great to know, thank you for the answer!